### ISSUE TITLE
Ensure That Sudo Isn't Used (CKV2_DOCKER_1): Incorrect check?

### ISSUE THREAD

**Issue by Sebbo94BY** [bridgecrewio/checkov#5578]
The pull request #3948 from @gruebel introduced the check `Ensure That Sudo Isn't Used` (`CKV2_DOCKER_1`), which has been approved by @rotemavni @PelegLi:

https://github.com/bridgecrewio/checkov/blob/74dc72b76b22b82f46e38263158f1393f1d68bab/checkov/dockerfile/checks/graph_checks/RunUsingSudo.yaml#L7-L12

The purpose of these checks is only to ensure, that the command(s), which get executed in the Dockerfile by `RUN` don't get executed by `sudo`, correct? Mainly because Docker automatically executes these commands already with `root` privileges, right?

So these should fail / alert:
```dockerfile
RUN sudo apt-get update
RUN  sudo apt-get update
RUN apt-get update && sudo apt-get install vim
RUN apt-get update \
             && sudo apt-get install vim
```

But these should NOT fail / alert:
```dockerfile
RUN usermod -aG sudo someOtherGroup
RUN usermod -aG \
             sudo \
             someOtherGroup
```

Is this correct?

If yes, the implementation of this check is currently incorrect. For the `usermod` example it's currently for example alerting / failing as it contains ` sudo `.

It also looks like as it won't match and alert these type of definitions:
```dockerfile
RUN ["sudo", "apt-get", "update"]
```

Did I missunderstood the purpose of this check or is it currently incorrect and not properly alerting on all cases?

**Comment by daltonv** [comment]
I agree this check is not correct. hadolint has a similar check but it is okay with sudo being installed. You just can't run sudo commands in a dockerfile.

**Comment by Saarett** [comment]
Hi @Sebbo94BY, @daltonv, your claim sounds reasonable. Do you think that removing this part from the check would do the trick?

https://github.com/bridgecrewio/checkov/blob/74dc72b76b22b82f46e38263158f1393f1d68bab/checkov/dockerfile/checks/graph_checks/RunUsingSudo.yaml#L7-L12

**Comment by Sebbo94BY** [comment]
@Saarett I think this would only partitally solve the issue. I think it's super hard to catch all cases due to the possible formattings in the Dockerfile.

My initial post contains some examples, which should succeed and fail. I believe, it would be good to test against these. Some unit tests would be good - does this project has unit tests?

Should FAIL:
```Dockerfile
RUN sudo apt-get update
RUN  sudo apt-get update
RUN apt-get update && sudo apt-get install vim
RUN apt-get update \
             && sudo apt-get install vim
RUN ["sudo", "apt-get", "update"]
```

Should SUCCEED:
```Dockerfile
RUN usermod -aG sudo someOtherGroup
RUN usermod -aG \
             sudo \
             someOtherGroup
RUN ["apt-get", "install", "sudo"]
```

**Comment by Saarett** [comment]
@Sebbo94BY We LOVE unit tests! :)

Would you like to contribute a fix? It would be very helpful!

- The relevant file containing the logic, as you already know, is located at `checkov/dockerfile/checks/graph_checks/RunUsingSudo.yaml`.
- The test case can be found at `tests/dockerfile/graph_builder/checks/test_yaml_policies.py`, line 38 (the method is `test_RunUsingSudo`).
- The test files are located at `tests/dockerfile/graph_builder/checks/resources/RunUsingSudo` (2 failing examples, 1 passing example).
